src/hardware/dac.rs

```diff
@@ -102,34 +100,39 @@ impl Dac {
     /// Set the DAC output to current on a channel.
     ///
     /// # Args
-    /// * `curr` - Set current in ampere
+    /// * `current` - Set current in ampere
     /// * `ch` - Thermostat output channel
-    pub fn set(&mut self, curr: f32, ch: Channel) -> Result<(), Bounds> {
-        let value = i_to_dac(curr);
-        if !(0..1 << 20).contains(&value) {
+    pub fn set(&mut self, current: f32, ch: Channel) -> Result<(), Bounds> {
+        // current to DAC word conversion
+        let v = (current * 10.0 * R_SENSE) + VREF_TEC;
+        let value = ((v * MAXCODE) / VREF_DAC) as u32;
```

Comment:
I think you want to rewrite this to use i32 here.
Otherwise your negative bounds check is questionable.

Reply:
Or more specifically, do the bounds check in `f32` before converting to `u32`.

Reply:
And if you just do `MAXCODE = 1 << 20`, and `!(0.0..MAXCODE as f32).contains()` then you also don't need the `<< 2` in L114.